tools/clang/lib/Sema/SemaHLSLDiagnoseTU.cpp

```diff
@@ -354,10 +379,14 @@ void hlsl::DiagnoseTranslationUnit(clang::Sema *self) {
     }
   }
 
-  CallGraphWithRecurseGuard callGraph;
+  const auto *shaderModel =
+      hlsl::ShaderModel::GetByName(self->getLangOpts().HLSLProfile.c_str());
+  DXIL::ShaderKind shaderKind = shaderModel->GetKind();
```

Comment:
How about `profileKind`, as it pertains to the `HLSLProfile` option, or perhaps just use `shaderModel->GetKind()` directly later, since keeping this as a separate local variable isn't really necessary.

Reply:
Fixed.